examples/02-plot/spherical.py

```diff
@@ -0,0 +1,226 @@
+
+    Parameters
+    ----------
+    lon: array-like
+        Array of longitudes of shape (M,) [degrees]
+    lat: array-like
+        Array of latitudes of shape (N,) [degrees]
+    levels: array-like
+        Array of vertical levels of shape (P,)
+
+    Returns
+    -------
+    pyvista.StructuredGrid
+    """
+    x, y, z = np.meshgrid(np.radians(lons), np.radians(lats), levels)
+    # Transform grid to cartesian coordinates
+    x_cart = levels * np.cos(y) * np.cos(x)
+    y_cart = levels * np.cos(y) * np.sin(x)
+    z_cart = levels * np.sin(y)
+    # Make a grid object
+    return pv.StructuredGrid(x_cart, y_cart, z_cart)
+
+
+def transform_vectors_sph_to_cart(lon, lat, levels, u, v, w):
+    """
+    Transform vectors from spherical coordinates (r, lat, lon) to cartesian coordinates (z, y, x).
+
+    Parameters
+    ----------
+    lon: array-like
+        Array of longitudes of shape (M,) [degrees]
+    lat: array-like
+        Array of latitudes of shape (N,) [degrees]
+    levels: array-like
+        Array of vertical levels of shape (P,)
+    u: array-like
+        Array of x-wind component (zonal wind) of shape (P, N, M)
+    v: array-like
+        Array of y-wind component (meridional wind) of shape (P, N, M)
+    w: array-like
+        Array of z-wind component (vertical wind) of shape (P, N, M)
+
+    Returns
+    -------
+    u_t, v_t, w_t: array-like
+        Arrays of transformed x-, y-, z-wind components, respectively.
+    """
+    xx, yy, _ = np.meshgrid(np.radians(lon), np.radians(lat), levels, indexing="ij")
+    x, y = xx.squeeze(), yy.squeeze()
+
+    # Transform wind components from spherical to cartesian coordinates
+    # https://en.wikipedia.org/wiki/Vector_fields_in_cylindrical_and_spherical_coordinates
+    #
+    # Note spherical coordinates are usually defined using a polar angle,
+    # while here `y` is latitude and `v` is reversed
+    u_t = np.cos(y) * np.cos(x) * w - np.sin(y) * np.cos(x) * v - np.sin(x) * u
+
+    v_t = np.cos(y) * np.sin(x) * w - np.sin(y) * np.sin(x) * v + np.cos(x) * u
+
+    w_t = np.sin(y) * w + np.cos(y) * v
+
+    return u_t, v_t, w_t
+
+
+# First, create some dummy data
+
+# Approximate radius of the Earth
+RADIUS = 6371.0
+
+# Longitudes and latitudes
+x = np.arange(0, 360, 5)
+y = np.arange(-90, 91, 10)
+
+xx, yy = np.meshgrid(x, y)
+
+
+# x- and y-components of the wind vector
+u_vec = np.cos(np.radians(xx))  # zonal
+v_vec = np.sin(np.radians(yy))  # meridional
+
+# Scalar data
+scalar = u_vec ** 2 + v_vec ** 2
+
+# Create arrays of grid cell boundaries, which have shape of (x.shape[0] + 1)
+xx_bounds = _cell_bounds(x)
+yy_bounds = _cell_bounds(y)
+# Vertical levels
+# in this case a single level slighly above the surface of a sphere
+levels = [RADIUS * 1.01]
+
+###############################################################################
+# Create a structured grid
+grid_scalar = grid_from_sph_coords(xx_bounds, yy_bounds, levels)
```

Comment:
This is interesting and something I may want to use down the road quite a bit.

I'd like to generalize this to be a single, multi-dimensional method. I see this being really useful when creating a structured grid from cell centers. It may be nice to have another utility that creates a structured grid leverage `_cell_bounds` internally. I'll think about this and maybe implement it down the road.

In the meantime, let's leave this here